### PR TITLE
Product description AI: add an entry point from the product form

### PR DESCRIPTION
**Pull request by jaclync**
Part of #9928

## Description

This PR added the second entry point to product description AI behind the i2 feature flag. The location in the product form is below the description row when:
- The product form is editable 
- Product description AI is enabled
- `isProductDescriptionAIFromStoreOnboarding` feature flag is enabled (only for debug & alpha builds)

I decided to include this logic in `DefaultProductFormTableViewModel` instead of `ProductFormActionsFactoryProtocol` because the description AI row visibility logic is shared for all `ProductFormActionsFactoryProtocol` implementations (products and variations). Please feel free to share any thoughts on this.

📝 The sparkle icon is currently an emoji, I'm confirming with the design to see if this is okay for the first iteration in NMncL2ViLjomfLhuyEhVM1-fi-219:52831.

### Subtle button style

A new subtle button style was also introduced in this PR, following the design NMncL2ViLjomfLhuyEhVM1-fi-236_86455.
Screenshot 2023-06-13 at 11 10 12 AM

It took me some time get the styles right in `UIButton+Helpers` using `UIButton.Configuration`. In other button styles, I noticed that the `UIButton.Configuration` isn't actually used for the button 🤔 (for example, [this configuration variable](https://github.com/woocommerce/woocommerce-ios/blob/8c390f577ee8d3315d3c5fe78e8bfa27a8618d99/WooCommerce/Classes/Extensions/UIButton%2BHelpers.swift#L11) isn't set to the button later on)

In `ButtonTableViewCell`, a new button style `subtle` is created. The xib was updated so that the constraint that pins the button's trailing edge to the superview can be disabled for the subtle button style (we want to fit to content).

### Product form action handling

Right now both entry points are presenting and dismissing the bottom sheet with duplicated code, I plan to refactor this navigation logic to a coordinator in a future PR since the PR is already big.

#### WPCOM site

Prerequisite: the site is hosted on WPCOM

- Go to the products tab
- Create a product if needed
- Tap on an editable product --> a button "Write it for me" should be shown below the description row
- Tap on `Write it for me` --> a bottom sheet should be shown, and an event should be tracked `🔵 Tracked product_description_ai_button_tapped, properties: [AnyHashable("source"): "product_form", AnyHashable("is_wpcom_store"): true, ...]`
- Enter some features if needed
- Tap `Generate` when it's ready
- Tap `Apply` when a description is generated --> the generated description should be set to the product description, and `Save` CTA should be shown in the product form navigation bar

#### Self-hosted site

Prerequisite: the site is not hosted on WPCOM

- Go to the products tab
- Create a product if needed
- Tap on an editable product --> a button "Write it for me" should not be shown below the description row, and the description row should look like before with a bottom separator

---
- [x] @jaclync tests when `productDescriptionAIFromStoreOnboarding` is disabled, the CTA isn't shown

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/1945542/468b6d4b-3a90-4d7c-8587-4ed0945e3697

\ | product form | with bottom sheet
-- | -- | --
light mode | Simulator Screen Shot - iPhone 14 Pro - 2023-06-14 at 09 31 40 | Simulator Screen Shot - iPhone 14 Pro - 2023-06-14 at 09 31 42
dark mode | Simulator Screen Shot - iPhone 14 Pro - 2023-06-14 at 09 31 30 | Simulator Screen Shot - iPhone 14 Pro - 2023-06-14 at 09 31 32

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by itsmeichigo** [comment]
> 📝 The sparkle icon is currently an emoji, I'm confirming with the design to see if this is okay for the first iteration

I think we can use the system image named `sparkles`, which looks pretty close to the icon in the design. I'm using it for the product sharing AI screen.

File: WooCommerce/Classes/Extensions/UIButton+Helpers.swift [hunk 1]
```diff
@@ -80,6 +80,32 @@ extension UIButton {
...
+    ///
+    func applySubtleButtonStyle() {
+        let configuration: UIButton.Configuration = {
+            var configuration = UIButton.Configuration.filled()
+            configuration.contentInsets = .init(
+                top: Style.verticalInset,
+                leading: Style.horizontalInset,
+                bottom: Style.verticalInset,
+                trailing: Style.horizontalInset
+            )
```
Comment (itsmeichigo) [inline WooCommerce/Classes/Extensions/UIButton+Helpers.swift]: Nit: These insets seem a bit larger than in the design, should we customize it for this style?

Screenshot 2023-06-13 at 12 56 26

**Review by itsmeichigo — APPROVED** [review]
LGTM 👍 

> I decided to include this logic in DefaultProductFormTableViewModel instead of ProductFormActionsFactoryProtocol because the description AI row visibility logic is shared for all ProductFormActionsFactoryProtocol implementations (products and variations). Please feel free to share any thoughts on this.

This also makes sense to me.

**Comment by jaclync** [comment]
> I think we can use the system image named `sparkles`, which looks pretty close to the icon in the design. I'm using it for the product sharing AI screen.

To match the design more accurately, I exported the sparkles icon from the design and configured the button further more to also update the content insets in https://github.com/woocommerce/woocommerce-ios/pull/9935/commits/0a75cd17509ba67a7af7605216232acce3f1e94a. Feel free to take a look, screenshots are updated in the PR description.

File: WooCommerce/Classes/Extensions/UIButton+Helpers.swift (hunk 1 above)
Comment (jaclync) [reply WooCommerce/Classes/Extensions/UIButton+Helpers.swift]: Updated in https://github.com/woocommerce/woocommerce-ios/pull/9935/commits/0a75cd17509ba67a7af7605216232acce3f1e94a

**Comment by jaclync** [comment]
Merging the PR now to avoid merge conflicts, please feel free to post any suggestions and I'll add them as subtasks to the issue.

**Comment by itsmeichigo** [comment]
Yeah it makes sense to export the icon, looks like we're using it in quite a few places now 😅.

**Pull request merged**